tests/components/habitica/test_config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Test the habitica config flow."""
+from asyncio import Future
+
+from homeassistant import config_entries, setup
+from homeassistant.components.habitica.config_flow import InvalidAuth
+from homeassistant.components.habitica.const import CONF_API_USER, DEFAULT_URL, DOMAIN
+from homeassistant.const import HTTP_OK
+
+from tests.async_mock import MagicMock, patch
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    request_mock = MagicMock()
+    type(request_mock).status_code = HTTP_OK
+
+    mock_obj = MagicMock()
+    mock_obj.user.get.return_value = Future()
+    mock_obj.user.get.return_value.set_result(None)
+
+    with patch(
+        "homeassistant.components.habitica.config_flow.HabitipyAsync",
+        return_value=mock_obj,
+    ), patch(
+        "homeassistant.components.habitica.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.habitica.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"api_user": "test-api-user", "api_key": "test-api-key"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Default username"
+    assert result2["data"] == {
+        "url": DEFAULT_URL,
+        "api_user": "test-api-user",
+        "api_key": "test-api-key",
+    }
+    await hass.async_block_till_done()
```

Comment:
Please move this block till done inside the patch context manager under the configure call.